charts/cp-kafka/values.yaml

```diff
@@ -46,9 +46,10 @@ configurationOverrides:
   ## - http://kafka.apache.org/documentation/#security_configbroker
   ## - https://cwiki.apache.org/confluence/display/KAFKA/KIP-103%3A+Separation+of+Internal+and+External+traffic
   ##
+  ## Advertised listeners will use the firstListenerPort value as it's default unless overridden here.
   ## Setting "advertised.listeners" here appends to "PLAINTEXT://${POD_IP}:9092,"
-  "advertised.listeners": |-
-   EXTERNAL://${HOST_IP}:$((31090 + ${KAFKA_BROKER_ID}))
+  # "advertised.listeners": |-
+  # EXTERNAL://${HOST_IP}:$((31090 + ${KAFKA_BROKER_ID}))
   "listener.security.protocol.map": |-
```

Comment:
I think you should also comment those 2 lines

Reply:
The advertised.listeners value from these two lines now get automatically added via $advertisedListenersBasedOnFirstListenerPort in the statefulset.yaml, the benefit of this approach is that we are no longer hardcoded to 31090 and $advertisedListenersBasedOnFirstListenerPort will instead use the firstListenerPort value instead (which is also defaulted to 31090). I have a special need (deploying kafka in different kubernetes namespaces)where i need to change the firstListenerPort from the default and have the advertised listeners also reflect this value.

Optionally you can overwrite the "advertised.listeners" from the helm command line and this would bypass the $advertisedListenersBasedOnFirstListenerPort and use whatever value is passed in at command line instead, so i think we don't loose anything by omitting these two lines.